saphir/src/server.rs

```diff
@@ -412,31 +391,65 @@ unsafe impl Send for Stack {}
 unsafe impl Sync for Stack {}
 
 impl Stack {
-    fn new_handler(&'static self, peer_addr: Option<SocketAddr>) -> StackHandler {
-        StackHandler { stack: self, peer_addr }
+    fn new_handler(&'static self, timeout_ms: Option<u64>, peer_addr: Option<SocketAddr>) -> StackHandler {
+        StackHandler {
+            timeout_ms,
+            stack: self,
+            peer_addr,
+        }
     }
 
-    async fn invoke(&self, mut req: Request<Body>) -> Result<Response<Body>, SaphirError> {
+    async fn invoke(&self, mut req: Request<Body>, timeout_ms: Option<u64>) -> Result<Response<Body>, SaphirError> {
+        use tokio::time::{timeout, Duration, Elapsed};
+
         let meta = self.router.resolve_metadata(&mut req);
+        let mut err_req = Request::clone_for_err(&req);
```

Comment:
Why would you clone the whole request for err instead of the HttpContext, HttpContext CloneForErr would simply clone the HandlerMetadata and the operationId. This would be faster than cloning the request.

Reply:
The HttpContext doesn't work without a request because of the deref impl, and the creation of HttpContext consume the request.